packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -517,13 +517,13 @@ export class MirrorNodeClient {
         return `${constants.CACHE_KEY.GET_CONTRACT}.valid.${contractIdOrAddress}`;
     }
 
-    public getIsValidContractCache(contractIdOrAddress): any {
+    public async getIsValidContractCache(contractIdOrAddress): Promise<any> {
         const cachedLabel = this.getIsValidContractCacheLabel(contractIdOrAddress);
-        return this.cache.get(cachedLabel, MirrorNodeClient.GET_CONTRACT_ENDPOINT);
+        return this.cacheService.get(cachedLabel, MirrorNodeClient.GET_CONTRACT_ENDPOINT);
     }
 
     public async isValidContract(contractIdOrAddress: string, requestIdPrefix?: string, retries?: number) {
-        const cachedResponse: any = this.getIsValidContractCache(contractIdOrAddress);
+        const cachedResponse: any = await this.getIsValidContractCache(contractIdOrAddress);
```

Comment:
Shouldn't be needed right?
```suggestion
        const cachedResponse: any = this.getIsValidContractCache(contractIdOrAddress);
```

Reply:
You're right, thanks.